content/blog/writing-better-angular-tests/index.md

```diff
@@ -0,0 +1,208 @@
+
+```json
+tsconfig.spec.json
+{
+  ...,
+  "files": [
+    "src/test.ts", <- delete
+    ...
+    ]
+}
+```
+
+The project also no longer needs the `test` key inside of `angular.json` and thus it can be removed.
+
+```json
+angular.json
+{
+  ...,
+  "test": {...} <- delete
+  ....
+}
+```
+
+Finally the project no longer needs the Jasmine types in the spec configuration
+
+```json
+tsconfig.spec.json
+{
+  ...,
+  "compilerOptions": {
+    ...,
+    "types": [
+      "jasmine", <- delete
+      ...
+    ]
+  }
+}
+```
+
+Now the project is ready for installing any other test runner.
+
+### Setting up Jest
+
+Now that the project has no Karma it can be setup with Jest
+
+#### Install Jest
+
+```bash
+npm i -D @types/jest jest jest-preset-angular ts-jest
+```
+
+This installs Jest, the types for Jest, a TypeScript pre-processor for Jest, and a preset that makes setting up Jest much easier.
+
+#### Configure Jest
+
+The project now needs to know how to best utilize Jest. Creating and modify the following files will allow Jest to load it's own configuration.
+
+```js
+jest.config.js
+module.exports = {
+    preset: 'jest-preset-angular',
+    setupFilesAfterEnv: [
+        '<rootDir>/jest.setup.ts'
+    ]
+};
+```
+
+```typescript
+jest.setup.ts
+import 'jest-preset-angular';
+```
+
+```json
+tsconfig.spec.json
+{
+  ...,
+  "compilerOptions": {
+    ...,
+    "types": [
+      "jest", <- new
+      ...
+    ]
+  }
+}
+```
+
+```json
+tsconfig.json
+{
+  ...,
+  "compilerOptions": {
+    ...,
+    "esModuleInterop": true, <- new
+    "emitDecoratorMetadata": true, <- new
+    ...
+  },
+  ...
+}
+```
+
+```json
+package.json
+{
+  ...,
+  "scripts": {
+    ...,
+    "test": "jest --coverage --config ./jest.config.js", <- new
+    "test:watch": "jest -o --watch --config ./jest.config.js", <- new
+    ...
+  },
+  ...
+}
+```
+
+Jest is now the test runner for the project. It can now be used to run the tests, but even better it can be used in combination with Testing Library.
+
+### Install Angular Testing Library
+
+Now the project is ready to have better tests written for it and by using [Angular Testing Library](https://testing-library.com/docs/angular-testing-library/intro) the tests can be simplified with some great helpers.
+
+```bash
+npm install --save-dev @testing-library/angular
+```
+
+# Ready, Steady, Test!
+
+Now that the project has a better testing library with some great helpers better tests can be written. There are plenty of [great examples](https://testing-library.com/docs/angular-testing-library/examples) for learning and [Tim Deschryver](https://timdeschryver.dev/blog/good-testing-practices-with-angular-testing-library) has more examples to help in that endeavor, and the Angular Testing Library will make tests much simpler to write and maintain. With Angular, good tests, and plenty of confidence anyone would be happy to ship a project with this setup.
+

```

Comment:
I like to end articles mentioning that if questions are present they can turn to our Discord server and come and "hang out" and such at this URL: https://discord.gg/FMcvc6T

This isn't required to be added but could help "scale down" the article in tone if you'd like :)

Reply:
discord... gitter... teams... whatever... but yes it can help tone it down. The question is do we want to poll and see about moving that to the layout instead of the blog posts?